tests/translations.py

```diff
@@ -66,110 +70,57 @@ def set_language(client: Client, lang: str):
-                break
-            data = data[lookup]
-        if language_data_missing:
-            continue
-        assert isinstance(data, str), f"{path} is not a string"
-        if template:
-            data = data.format(*template)
-        texts.append(data)
-
-    if lower:
-        texts = [t.lower() for t in texts]
-    texts = [t.replace("\xa0", " ").strip() for t in texts]
-    return texts
-
-
-def assert_equals(text: str, path: str, template: t.Iterable[t.Any] = ()) -> None:
-    # TODO: we can directly pass in the current device language
-    texts = _resolve_path_to_texts(path, template)
-    assert text.lower() in texts, f"{text} not found in {texts}"
-
-
-def assert_equals_multiple(
-    text: str, paths: list[str], template: t.Iterable[t.Any] = ()
-) -> None:
-    texts: list[str] = []
-    for path in paths:
-        texts += _resolve_path_to_texts(path, template)
-    assert text.lower() in texts, f"{text} not found in {texts}"
-
-
-def assert_in(text: str, path: str, template: t.Iterable[t.Any] = ()) -> None:
-    texts = _resolve_path_to_texts(path, template)
-    for tt in texts:
-        if tt in text.lower():
-            return
-    assert False, f"{text} not found in {texts}"
-
-
-def assert_in_multiple(
-    text: str, paths: list[str], template: t.Iterable[t.Any] = ()
-) -> None:
-    texts: list[str] = []
-    for path in paths:
-        texts += _resolve_path_to_texts(path, template)
-    for tt in texts:
-        if tt in text.lower():
-            return
-    assert False, f"{text} not found in {texts}"
-
-
-def assert_startswith(text: str, path: str, template: t.Iterable[t.Any] = ()) -> None:
-    texts = _resolve_path_to_texts(path, template)
-    for tt in texts:
-        if text.lower().startswith(tt):
-            return
-    assert False, f"{text} not found in {texts}"
-
-
-def assert_template(text: str, template_path: str) -> None:
-    templates = _resolve_path_to_texts(template_path)
-    for tt in templates:
-        # Checking at least the first part
-        first_part = tt.split("{")[0]
-        if text.lower().startswith(first_part):
-            return
-    assert False, f"{text} not found in {templates}"
-
-
-def translate(
-    path: str, template: t.Iterable[t.Any] = (), lower: bool = False
-) -> list[str]:
-    # Do not converting to lowercase, we want the exact value
-    return _resolve_path_to_texts(path, template, lower=lower)
+class Translation:
+    FORMAT_STR_RE = re.compile(r"\\{\d+\\}")
+
+    def __init__(self, lang: str) -> None:
+        self.lang = lang
+        self.lang_json = get_lang_json(lang)
+
+    @property
+    def translations(self) -> dict[str, str]:
+        return self.lang_json["translations"]
+
+    def _translate_raw(self, key: str) -> str:
+        tr = self.translations.get(key)
+        if tr is not None:
+            return tr
+        if self.lang != "en":
+            return TRANSLATIONS["en"]._translate_raw(key)
```

Comment:
This branch returns `"en"` version in case that the `key` was not found in the set language. 

How about printing a warning here? Something like `"Warning: {key} not found in {self.lang}"`

Reply:
like this? f9218ae59ec5